providers/BaseProviders/src/net/ess3/provider/SpawnerProvider.java

```diff
@@ -1,17 +1,19 @@
-package net.ess3.nms;
+package net.ess3.provider;
 
 import com.google.common.collect.ImmutableMap;
-import net.ess3.providers.Provider;
 import org.bukkit.ChatColor;
-import org.bukkit.Material;
 import org.bukkit.entity.EntityType;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.ItemMeta;
 
 import java.util.Map;
 
-public abstract class SpawnerProvider implements Provider {
-    private Map<EntityType, String> entityToDisplayName = ImmutableMap.<EntityType, String>builder()
+public interface SpawnerProvider extends Provider {
+    ItemStack setEntityType(ItemStack is, EntityType type) throws IllegalArgumentException;
+
+    EntityType getEntityType(ItemStack is) throws IllegalArgumentException;
+
+    Map<EntityType, String> entityToDisplayName = ImmutableMap.<EntityType, String>builder()
```

Comment:
Pretty sure private was removed here on accident?
```suggestion
    private Map<EntityType, String> entityToDisplayName = ImmutableMap.<EntityType, String>builder()
```

Reply:
Removed due to the fact that interfaces cannot hold private modifiers.